src/provider/grpcProvider/grpcProvider.ts

```diff
@@ -0,0 +1,281 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import { Mas } from '../../basicElements'
+import { Operation } from '../../operation'
+import { SmartContract } from '../../smartContracts'
+import { Provider, ReadSCData, ReadSCParams, SignedData } from '..'
+import { Account } from '../../account'
+import { GrpcPublicProvider } from './grpcPublicProvider'
+import { GrpcApiUrl } from '../../utils/networks'
+import {
+  FunctionCall,
+  ReadOnlyExecutionCall,
+  ScExecutionEvent,
+  ScExecutionEventStatus,
+} from '../../generated/grpc/massa/model/v1/execution_pb'
+import {
+  AddressBalanceCandidate,
+  AddressBalanceFinal,
+  ExecuteReadOnlyCallRequest,
+  ExecutionQueryRequestItem,
+  QueryStateRequest,
+} from '../../generated/grpc/public_pb'
+import { NativeAmount } from '../../generated/grpc/massa/model/v1/amount_pb'
+import { PublicServiceClient } from '../../generated/grpc/PublicServiceClientPb'
+import { SCOutputEvent } from '../../generated/client-types'
+
+/**
+ * GrpcProvider implements the Provider interface using gRPC for Massa blockchain interactions
+ */
+export class GrpcProvider extends GrpcPublicProvider implements Provider {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  private readonly _providerName: string = 'Massa GRPC provider'
+
+  /**
+   * Creates a new GrpcProvider instance
+   * @param url - The gRPC endpoint URL
+   * @param account - The account associated with this provider
+   */
+  constructor(
+    public client: PublicServiceClient,
+    public url: string,
+    private account: Account
+  ) {
+    super(client, url)
+  }
+
+  static buildnet(account: Account): GrpcProvider
+  static buildnet(): GrpcPublicProvider
+  static buildnet(account?: Account): GrpcProvider | GrpcPublicProvider {
+    return account
+      ? GrpcProvider.fromGrpcUrl(GrpcApiUrl.Buildnet, account)
+      : GrpcProvider.fromGrpcUrl(GrpcApiUrl.Buildnet)
```

Comment:
just `return GrpcProvider.fromGrpcUrl(GrpcApiUrl.Buildnet, account)`should work

Reply:
nop